### PR TITLE
feat: add removeAllMessages function in react nc

### PR DESCRIPTION
**Pull request by jainpawan21**
### What change does this PR introduce?
- Add `useRemoveAllNotification` hook
- Add `removeAllMessages` function in `useNotifications` hook
- Update API reference docs for above

### CODE REVIEW

File: packages/notification-center/src/hooks/useRemoveAllNotifications.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { useMutation, useQueryClient, UseMutationOptions } from '@tanstack/react-query';
+import type { IMessage } from '@novu/shared';
+
+import { useNovuContext } from './useNovuContext';
+import { useFetchNotificationsQueryKey } from './useFetchNotificationsQueryKey';
+
+interface IRemoveAllNotificationsVariables {
```
Comment (p-fernandez) [inline packages/notification-center/src/hooks/useRemoveAllNotifications.ts:7]:
```suggestion
interface IRemoveAllNotificationsPayload {
```

File: packages/notification-center/src/hooks/useRemoveAllNotifications.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  feedId?: string;
```
Comment (p-fernandez) [inline packages/notification-center/src/hooks/useRemoveAllNotifications.ts:8]: If we make this optional then the mutation code should be prepared for it, checking if the `feedId` is being set and executing the mutation if so (line 23).

File: packages/client/src/api/api.service.ts [hunk 3]
```diff
@@ -70,6 +70,14 @@ export class ApiService {
     return await this.httpClient.delete(`/widgets/messages/${messageId}`, {});
   }
 
+  async removeAllMessages(feedId?: string): Promise<any> {
+    const url = feedId
+      ? `/widgets/messages?feedId=${feedId}`
+      : `/widgets/messages`;
```
Comment (p-fernandez) [inline packages/client/src/api/api.service.ts:76]: This is dangerous, to allow to delete all messages if a feedId is missed for any reason.

**Review by davidsoderberg — APPROVED** [review]
Looks good to me :)

**Review by p-fernandez — COMMENTED** [review]
I am missing the context of what needs to be done and why to be able to have a full input on this as at first sight there are 2 functionalities implemented: removing all notifications and removing all notifications from a feed. Could you provide more context? 🙏🏻

File: packages/client/src/api/api.service.ts (hunk 3 above)
Comment (jainpawan21) [reply packages/client/src/api/api.service.ts:76]: feedId is optional, user can remove either feed specific message or all messages

File: docs/docs/notification-center/react/api-reference.md [hunk 4]
```diff
@@ -387,6 +388,7 @@ interface INotificationsContext {
 | markAllNotificationsAsRead     | function                                                    | The function allowing you to mark all notifications as read.                                                                  |
 | markAllNotificationsAsSeen     | function                                                    | The function allowing you to mark notifications as seen.                                                                      |
 | removeMessage                  | function                                                    | The function allowing you to delete a notification.                                                                           |
+| removeAllMessages              | function                                                    | The function allowing you to delete all feeds notifications or specific feed's all notification at once.                      |
```
Comment (p-fernandez) [inline docs/docs/notification-center/react/api-reference.md:391]:
```suggestion
| removeAllMessages              | function                                                    | The function allowing you to delete all notifications or a specific feed's notifications at once.                      |
```

**Review by p-fernandez — APPROVED** [review]

**Pull request merged**